cmake/Modules/MacroQbtCommonConfig.cmake

```diff
@@ -0,0 +1,106 @@
+# Set common variables and create some interface-only library targets
+# that some or all other targets will link to, either directly or transitively,
+# to consume common compile options/definitions
+
+macro(qbt_common_config)
+
+    # treat value specified by the CXX_STANDARD target property as a requirement by default
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+    # these definitions are only needed for calls to
+    # lt::generate_fingerprint and for the qbittorrent.rc file on Windows
+    add_library(qbt_version_definitions INTERFACE)
+
+    target_compile_definitions(qbt_version_definitions INTERFACE
+        QBT_VERSION_MAJOR=${qBittorrent_VERSION_MAJOR}
+        QBT_VERSION_MINOR=${qBittorrent_VERSION_MINOR}
+        QBT_VERSION_BUGFIX=${qBittorrent_VERSION_PATCH}
+        QBT_VERSION_BUILD=${qBittorrent_VERSION_TWEAK}
+    )
+
+    add_library(qbt_common_cfg INTERFACE)
+
+    # Full C++ 14 support is required
+    target_compile_features(qbt_common_cfg INTERFACE
+        cxx_std_14
+        cxx_contextual_conversions
+        cxx_binary_literals
+        cxx_decltype_auto
+        cxx_lambda_init_captures
+        cxx_generic_lambdas
+        cxx_variable_templates
+        cxx_relaxed_constexpr
+        cxx_attribute_deprecated
+        cxx_digit_separators
+    )
+
+    set(QBT_FULL_VERSION "${qBittorrent_VERSION}${QBT_VER_STATUS}")
+
+    target_compile_definitions(qbt_common_cfg INTERFACE
+        QBT_VERSION="v${QBT_FULL_VERSION}"
+        QBT_VERSION_2="${QBT_FULL_VERSION}"
+        QT_DEPRECATED_WARNINGS
+        QT_NO_CAST_TO_ASCII
+        QT_NO_CAST_FROM_BYTEARRAY
+        QT_USE_QSTRINGBUILDER
+        QT_STRICT_ITERATORS
+        # GENEXP: the $<CONFIG:...> genexp is the only robust way of providing
+        # build type-dependent functionality for both single config and multi config generators alike
+        $<$<NOT:$<CONFIG:Debug>>:QT_NO_DEBUG_OUTPUT>
+    )
+
+    if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
+        target_compile_definitions(qbt_common_cfg INTERFACE
+            NTDDI_VERSION=0x06010000
+            _WIN32_WINNT=0x0601
+            _WIN32_IE=0x0601
+            UNICODE
+            _UNICODE
+            WIN32_LEAN_AND_MEAN
+            _CRT_SECURE_NO_DEPRECATE
+            _SCL_SECURE_NO_DEPRECATE
```

Comment:
Can you try compiling without the above 2 macros? If removing them does not produce additional compile warnings/errors then please remove them, as they are not present in winconf.pri either.

Reply:
Removed, compiled with no errors or warnings.

I also learned the following:

- It is not necessary to specify the value of `_WIN32_IE`, so I removed that, too. It is automatically set to the correct value in the `sdkddkver.h` Windows header based on the version set in `_WIN32_WINNT`.

- With `_WIN32_WINNT` defined, it is not necessary to define `WIN_VER`, it gets set automatically in the same header.

- `NTDDI_VERSION` is useful for finer grained version control, but currently it does nothing, since there is no finer grain control possibie for Windows 7 versions. It will come in handy when we need to distinguish among Windows 10 versions.

References:
https://docs.microsoft.com/en-us/windows/win32/WinProg/using-the-windows-headers
https://docs.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt

- `UNICODE` and `_UNICODE` should both be defined (as they are now)

References:
https://docs.microsoft.com/en-us/cpp/text/unicode-programming-summary
https://docs.microsoft.com/en-us/windows/win32/learnwin32/working-with-strings

> Be careful: Some headers use the preprocessor symbol `UNICODE`, others use `_UNICODE` with an underscore prefix. Always define both symbols. Visual C++ sets them both by default when you create a new project.

Reply:
>Removed, compiled with no errors or warnings.

OK.

>It is not necessary to specify the value of _WIN32_IE, so I removed that, too. It is automatically set to the correct value in the sdkddkver.h Windows header based on the version set in _WIN32_WINNT.

You shouldn't do more changes than asked here. Now you deviate from winconf.pri again, if you want `_WIN32_IE` (or anything) removed, you should proposed in another PR.

Reply:
@Chocobo1 Alright, fixed.